packages/camera/camera_android_camerax/android/src/main/java/io/flutter/plugins/camerax/CameraSelectorHostApiImpl.java

```diff
@@ -0,0 +1,70 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugins.camerax;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.VisibleForTesting;
+import androidx.camera.core.CameraInfo;
+import androidx.camera.core.CameraSelector;
+import io.flutter.plugin.common.BinaryMessenger;
+import io.flutter.plugins.camerax.GeneratedCameraXLibrary.CameraSelectorHostApi;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CameraSelectorHostApiImpl implements CameraSelectorHostApi {
+  private final BinaryMessenger binaryMessenger;
+  private final InstanceManager instanceManager;
+
+  @VisibleForTesting public CameraSelector.Builder cameraSelectorBuilder;
```

Comment:
Optional:

So having a settable parameter like this is pretty typical for testing the `create` methods in host apis. However, most classes don't use builders and you will have to test the constructor. It's worth considering using a proxy class to handle all your class instantiations and static methods. For example 

```java
class CameraXProxy {
  CameraSelector createCameraSelector(lensFacing) {
   // build and return
  }

  ACameraXClass createACameraXClass() {
    return new ACameraXClass();
  }

  void staticMethodForACameraXClass() {
    ACameraXClass.staticMethod();
  }
}
```

And then you can add this value to any `HostApi` that has a `create` method. And mock it in tests.

You could also create a separate proxy for each host api. (e.g. `CameraSelectorProxy`).